### DISCUSSION TITLE
MAUI Android Flick/pan duration and animation

### DISCUSSION THREAD

**Discussion by angelinn** [Mapsui/Mapsui#2821] [Q&A]
Hi everybody.

During my previous performance investigations comparing different map applications in Android, I noticed that when you move/pan/flick the map in Mapsui:
* the movement function stops very suddenly - meaning when you let go of the finger the map goes very little distance by itself.  Making a smoother ending is better looking.
* the distance traveled (sensitivity) of the flick is very little, meaning when you swipe your finger the distance traveled is less than expected. This could be correlated to the my other point above. I was digging through what I can change and noticed that if I modify `MapControl.UpdateInterval` to a lower value - the distance traveled increases, but the movement of the map still stops very quickly.

Google Maps and [OsmDroid](https://play.google.com/store/apps/details?id=org.osmdroid&hl=en) are good examples of the behavior I'm looking for. A smoother curve for ending of the movement and more distance traveled per flick.

I understand both of these are personal choices, in my opinion they'd look better if they behaved more like the apps mentioned above.  If you'd like to reproduce the behavior, just swipe across Google Maps and a Mapsui Android app to see the difference.

Let me know if there are any configuration/source code tweaks I can make.

**Comment by pauldendulk** [comment]
This is not something I will have time for in the near future, but it makes sense. 

This is my idea about how this could be done:
- Make the FlingTracker een public property on the MapControl. Configuration changes should be made to the FlingTracker, not the MapControl.
- Add a way to allow modifications to the onFling threshold. Currently this is this simple check: `if (Math.Abs(velocityX) <= 200 && Math.Abs(velocityY) <= 200)`.
- The FlingTracker now calls the Navigator.Fling method. Inside that method the animation is created. If we want to make the animation function configurable it is better to move it to the FlingTracker. And on the FlingTracker add a way to override the method to create the animation.

Exactly how it it should be made configurable I don't know yet. It is tempting to add simple public properties for the fields that we currently use. But I can imagine that we could have an entirely different way to determine the threshold and the fling animation. But my main point here is that it should be part of the FlingTracker.
